common/app/templates/inlineJS/blocking/applyAdfreeRenderCondition.scala.js

```diff
@@ -0,0 +1,119 @@
+@()
+
+(function applyAdfreeRenderCondition(commercial){
+    var AD_FREE_COOKIE = 'gu_adfree_test',
+        MVT_ID_COOKIE = 'GU_mvt_id',
+        FIRST_TIME_VISIT_KEY = 'gu.firstVisitTime',
+        ALREADY_VISITED_KEY = 'gu.alreadyVisited';
+
+    if ((readLocalStorage(ALREADY_VISITED_KEY) || 0) == 0) {
+        writeLocalStorage(FIRST_TIME_VISIT_KEY, new Date().getTime());
+    }
+
+    if (isFirst3VisitDays() && isInVariantGroup()){
+        commercial.showingAdfree = true;
+        return 'is-adfree';
+    } else {
+        commercial.showingAdfree = false;
+        return '';
+    }
+
+    function isFirst3VisitDays() {
+        var firstVisit = readLocalStorage('gu.firstVisitTime') || 0;
+        return (new Date().getTime() - firstVisit) < 259200000;
+    }
+
+    function isInVariantGroup() {
+        var cookie = readCookie(AD_FREE_COOKIE);
+        if (cookie == null) {
+            assignUser();
+        }
+        return readCookie(AD_FREE_COOKIE) == "variant";
+    }
+
+    function assignUser() {
```

Comment:
It's annoying that we have to copy the A/B scripts into our preflight code, but I can't see an alternative, because we have to choose what the user will see before their first pageview renders: if the user is subject to the test, we must not show them any ads; if the user is not subject to the test, we cannot default them to seeing no adverts without compromising our revenue.

I think this is fair enough for an A/B test - it's not that much code, after all - but I think we'll want to take another look at this if and when this experiment graduates into something long-term.